### PR TITLE
Incorperate feedback on Lazy helpers

### PR DESCRIPTION
**Pull request by josephlr**
In #52 @RalfJung noted that some improvements could be make to the implementations of `LazyUsize` and `LazyFd`. This PR implements those changes.

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -65,6 +65,7 @@ impl LazyUsize {
             match self.0.compare_and_swap(Self::UNINIT, Self::ACTIVE, Relaxed) {
                 Self::UNINIT => {
                     let val = init();
+                    assert!(val != Self::ACTIVE);
```
Comment (newpavlov) [inline src/util.rs]: As I wrote earlier in #52 I would prefer to use code like this:
```rust
let mut val = init();
if val == Self::ACTIVE { val = Self::UNINIT }
self.0.store(val, Relaxed);
```

Comment (josephlr) [reply src/util.rs]: Any particular reason why? Wouldn't having `init()` returning `ACTIVE` be a bug? Or are you concerned with the `LazyUsize` getting "stuck" in the `ACTIVE` state after a panic?

Comment (newpavlov) [reply src/util.rs]: Panicking pulls formatting machinery (unless some careful setup is done) and it results in a larger binary. I haven't thought about infinite loop (e.g. if user will catch the panic), but it's indeed another drawback.

I think we both agree that it's extremely unlikely that `init` will return `Self::ACTIVE`, so it does not really matter.

Comment (josephlr) [reply src/util.rs]: I removed the assert and updated the docs. I used a slightly different implementation than above to maintain the invariant that `sync_init()` always returns the value of some run of `init()`.

File: src/util.rs [hunk 2]
```diff
@@ -65,8 +66,13 @@ impl LazyUsize {
             match self.0.compare_and_swap(Self::UNINIT, Self::ACTIVE, Relaxed) {
                 Self::UNINIT => {
                     let val = init();
-                    assert!(val != Self::ACTIVE);
-                    self.0.store(val, Relaxed);
+                    self.0.store(
+                        match val {
+                            Self::UNINIT | Self::ACTIVE => Self::UNINIT,
+                            val => val,
+                        },
```
Comment (newpavlov) [inline src/util.rs]: I think the following alternative will be a bit easier to read:
```rust
let val = match init() {
    Self::UNINIT | Self::ACTIVE => Self::UNINIT,
    val => val,
};
self.0.store(val, Relaxed);
```

Comment (newpavlov) [reply src/util.rs]: Ah, nevermind. I see your comment earlier.

**Pull request merged**